src/commands/post.ts

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable jsdoc/require-jsdoc */
+import {
+  ChannelType,
+  SlashCommandBuilder,
+  SlashCommandSubcommandBuilder,
+} from "discord.js";
+
+import { Command } from "../interfaces/commands/Command";
+import { CommandHandler } from "../interfaces/commands/CommandHandler";
+import { errorEmbedGenerator } from "../modules/commands/errorEmbedGenerator";
+import { beccaErrorHandler } from "../utils/beccaErrorHandler";
+
+import { handleInvalidSubcommand } from "./subcommands/handleInvalidSubcommand";
+import { handleCreate } from "./subcommands/post/handleCreate";
+import { handleDelete } from "./subcommands/post/handleDelete";
+import { handleEdit } from "./subcommands/post/handleEdit";
+
+const handlers: { [key: string]: CommandHandler } = {
+  create: handleCreate,
+  edit: handleEdit,
+  delete: handleDelete,
+};
+
+export const post: Command = {
+  data: new SlashCommandBuilder()
+    .setName("post")
+    .setDescription("Commands for managing server posts.")
+    .setDMPermission(false)
+    .addSubcommand(
+      new SlashCommandSubcommandBuilder()
+        .setName("create")
+        .setDescription("Command to create server post.")
+        .addChannelOption((option) =>
+          option
+            .setName("channel")
+            .setDescription("The channel to send this message in.")
+            .addChannelTypes(
+              ChannelType.GuildText,
+              ChannelType.GuildNews,
+              ChannelType.GuildNewsThread,
+              ChannelType.GuildPublicThread,
+              ChannelType.GuildPrivateThread
```

Comment:
```suggestion
              ChannelType.GuildText,
              ChannelType.GuildAnnouncement,
              ChannelType.AnnouncementThread,
              ChannelType.PublicThread,
              ChannelType.PrivateThread,
```
Replace with the newer type names.